### PR TITLE
Update App.jsx

### PR DESCRIPTION
**Pull request by uday-appdirect**
Closes #53

👨‍💻 Changes proposed
Use of typeof window !== 'undefined' creates an error. Refer https://nextjs.org/docs/messages/react-hydration-error
Restructured the useWindowDimensions hook and used the same logic inside useEffect so that this logic runs only on client
✔️ Check List (Check all the applicable boxes)
 My code follows the code style of this project.

 This PR does not contain plagiarized content.

 The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -21,6 +21,25 @@ const App = () => {
...
+  //     issue_number: pull_request.number,
+  //     body: 'Thank you for submitting a pull request! We will
+  //           try to review this as soon as we can.'
+  //   });
+  // }
+   
+  // run();
+
   const getText = async (url) => {
     let config = {
```
Comment (mrcoder991) [inline src/App.jsx:44]: Please add more information here, and fix this typo

**Review by mrcoder991 — CHANGES_REQUESTED** [review]
This is close to perfect! Please address the suggested inline change.

File: src/App.jsx (hunk 1 above)
Comment (udaygirhepunje) [inline src/App.jsx:48]: THIS IS A REVIEW COMMENT

**Review by mrcoder991 — CHANGES_REQUESTED** [review]
THIS IS A REVIEW BODY

File: src/App.jsx (hunk 1 above)
Comment (udaygirhepunje) [inline src/App.jsx:48]: THIS IS A REVIEW COMMENT

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY

File: src/App.jsx [hunk 2]
```diff
@@ -21,6 +21,29 @@ const App = () => {
     return response;
   }
```
Comment (udaygirhepunje) [inline src/App.jsx:23]: THIS IS A REVIEW COMMENT

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY

File: src/App.jsx [hunk 3]
```diff
@@ -21,6 +21,29 @@ const App = () => {
...
 
+  // const core = require('@actions/core');
+  // const github = require('@actions/github');
+  // const { context } = require('@actions/github')
```
Comment (udaygirhepunje) [inline src/App.jsx:26]: THIS IS A REVIEW COMMENT 6

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY 6

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY 6

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY 6

File: src/App.jsx (hunk 3 above)
Comment (udaygirhepunje) [inline src/App.jsx:26]: The initialization of `hasWindow` inside the useEffect ensures that `window` is only referenced in a client-side environment, which helps avoid React hydration errors. Good job!

Comment (udaygirhepunje) [inline src/App.jsx:26]: The initial state with `initDimensions` avoids having an undefined state on the server side. This effectively prevents  potential issues during server-side rendering (SSR). It also simplifies initial rendering.

Comment (udaygirhepunje) [inline src/App.jsx:26]: There's no need to pass the `hasWindow` flag continuously in the dependency array, as it’s only required in the initial check for the window object. Therefore, the effect can run once when the component is mounted.

**Review by mrcoder991 — COMMENTED** [review]
THIS IS A REVIEW BODY 6

**Pull request closed without merging**